src/types/note_model.ts

```diff
@@ -0,0 +1,18 @@
+import { getPredifinedUsers } from "../notes/get_all_users";
+import { User } from "./user_model";
+
+export class Note{
+    private note_id : Number;
+    private date : String;
+    private text : String;
+    private users : User[];
+
+    private notes : Note[] | undefined;
+    
+    constructor(note_id: Number, date: String, text: String, user: User[]){
+        this.note_id = note_id;
+        this.date = date;
+        this.text = text;
+        this.users = getPredifinedUsers();
```

Comment:
Can you please explain this? Looking at the constructor, for each note that we define, we are having multiple users. Is it so that a note created by a user can be modified by other users, so we keep an array of notes?

Reply:
Yes, If you want we can make it 1 : 1 for User to Note